README.md

```diff
@@ -161,6 +168,52 @@ Autoload can be customised using the following options:
   // routes can now be added to /defaultPrefix/something
   ```
 
+- `autoHooks` (optional) - Apply hooks from `autohooks.js` file(s) to plugins found in folder
+
+  Automatic hooks from `autohooks.js` files will be encapsulated with plugins. If `false`, all `autohooks.js` files will be ignored.
+
+  ```js
+  fastify.register(autoLoad, {
+    dir: path.join(__dirname, 'plugins'),
+    autoHooks: true // apply hooks to routes in this level
+  })
+  ```
+
+- `autoHooksPattern` (optional) - Regex to override the `autohooks` naming convention
+
+  ```js
+  fastify.register(autoLoad, {
+    dir: path.join(__dirname, 'plugins'),
+    autoHooks: true,
+    autoHooksPattern: /^[_.]?auto_?hooks(\.ts|\.js|\.cjs|\.mjs)$/i
+  })
+  ```
+
+- `cascadeHooks` (optional) - If using `autoHooks`, cascade hooks to all children. Ignored if `autoHooks` is `false`.
+
+  Default behaviour of `autoHooks` is to apply hooks only to the level on which the `autohooks.js` file is found. Setting `cascadeHooks: true` will continue applying the hooks to any children.
```

Comment:
Is this wrapping the hook in `fastify-plugin`?

Reply:
No, it's adding `skip-override: true` to the hook plugin(s) and encapsulating them with the routes, which are added to the `fastify` / `app` instance where the `fastify-autoload` plugin is registered.

Reply:
A note about this being implemented via Fastify's encapsulation feature and a link to that documentation would be good.

Reply:
Sounds good, which is the preferred link to the documentation?

https://www.fastify.io/docs/latest/Encapsulation/

or 

https://github.com/fastify/fastify/blob/master/docs/Encapsulation.md

Reply:
I'd go with the first one.

Reply:
I changed the wording to reflect that it's encapsulating files and added the link, does that work a bit better?

I also added a `routes.js` to the example tree in the autoHooks section, so it was more obvious that it was a plugin being loaded by `fastify-autoload` but unaffected by the hooks.